File: php/divergent-modifications/good/UserRepository.php
<?php

class UserRepository {
    private $db;

    public function __construct($db) {
        $this->db = $db;
    }

    public function updateUserProfile(int $userId, string $name, string $email): void {
        $stmt = $this->db->prepare("UPDATE users SET name = ?, email = ? WHERE id = ?");
        $stmt->execute([$name, $email, $userId]);
    }

    public function getUserEmail(int $userId): ?string {
        $stmt = $this->db->prepare("SELECT email FROM users WHERE id = ?");
        $stmt->execute([$userId]);
        $result = $stmt->fetch();
        return $result['email'] ?? null;
    }
}

File: python/divergent-modifications/good/user_repository.py
class UserRepository:
    def __init__(self, db):
        self.db = db

    def update_user_profile(self, user_id: int, name: str, email: str) -> None:
        cursor = self.db.cursor()
        cursor.execute("UPDATE users SET name = ?, email = ? WHERE id = ?", (name, email, user_id))
        self.db.commit()

    def get_user_email(self, user_id: int) -> str:
        cursor = self.db.cursor()
        cursor.execute("SELECT email FROM users WHERE id = ?", (user_id,))
        result = cursor.fetchone()
        return result[0] if result else None
